src/jit/codegenarm.cpp

```diff
@@ -156,7 +197,42 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
 
         case GT_CNS_DBL:
         {
-            NYI("GT_CNS_DBL");
+            GenTreeDblCon* dblConst   = tree->AsDblCon();
+            double         constValue = dblConst->gtDblCon.gtDconVal;
+            // TODO-ARM-CQ: Do we have a faster/smaller way to generate 0.0 in thumb2 ISA ?
+            if (targetType == TYP_FLOAT)
+            {
+                // Get a temp integer register
+                regMaskTP tmpRegMask = tree->gtRsvdRegs;
+                regNumber tmpReg     = genRegNumFromMask(tmpRegMask);
+                noway_assert(tmpReg != REG_NA);
+
+                float f = forceCastToFloat(constValue);
+                genSetRegToIcon(tmpReg, *((int*)(&f)));
+                getEmitter()->emitIns_R_R(INS_vmov_i2f, EA_4BYTE, targetReg, tmpReg);
+            }
+            else
+            {
+                assert(targetType == TYP_DOUBLE);
+
+                size_t* cv = (size_t*)&constValue;
```

Comment:
I would use `unsigned` here rather than `size_t`, as it is less likely to vary with the host architecture.